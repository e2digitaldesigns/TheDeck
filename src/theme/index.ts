export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
