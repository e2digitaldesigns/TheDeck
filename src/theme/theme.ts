export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
