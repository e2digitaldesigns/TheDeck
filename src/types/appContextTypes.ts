export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
