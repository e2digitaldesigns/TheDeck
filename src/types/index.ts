/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
