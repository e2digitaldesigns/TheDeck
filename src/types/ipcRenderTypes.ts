export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
