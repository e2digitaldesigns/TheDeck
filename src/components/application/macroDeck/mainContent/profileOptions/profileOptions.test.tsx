import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ProfileOptions from "./profileOptions";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../theme/theme";
import initGlobalState from "../../../../../__mock_data__/globalState.json";
import initAppState from "../../../../../__mock_data__/appState.json";
import _cloneDeep from "lodash/cloneDeep";

import { IntAppData, IntGlobalData } from "../../../../../types";

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

const mockAppState: IntAppData = {
  appState: _cloneDeep(initAppState),
  setAppState: jest.fn(appState => appState)
};

jest.mock("../../../../../hooks", () => ({
  __esModule: true,
  useAppData: () => mockAppState,
  useGlobalData: () => mockGlobalState,
  useObj: () => ({
    pageObj: jest.fn()
  }),
  usePage: () => ({
    activatePage: jest.fn(),
    createPage: jest.fn(),
    deletePage: jest.fn(),
    readPages: jest.fn()
  }),
  useProfile: () => ({
    readProfile: () => mockGlobalState.state.profiles[1],
    updateProfile: jest.fn()
  })
}));

const testSetup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ProfileOptions />
    </ThemeProvider>
  );
};

beforeEach(() => {
  mockAppState.appState = _cloneDeep(initAppState);
});

afterEach(() => {
  jest.clearAllMocks();
});

describe("<ProfileOptions/>", () => {
  it("Should render without errors", () => {
    mockAppState.appState.active.profileId =
      mockGlobalState.state.profiles[1]._id;
    const wrapper = testSetup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should show application", () => {
    mockAppState.appState.active.profileId =
      mockGlobalState.state.profiles[1]._id;
    const wrapper = testSetup();
    const component = wrapper.getByTestId(
      "profile_options__page-header-wrapper"
    );
    expect(component).toBeTruthy();
  });

  it("Should NOT show application", () => {
    const wrapper = testSetup();
    const component = wrapper.getByTestId("profile_options__not-loaded");
    expect(component).toBeTruthy();
  });
});
