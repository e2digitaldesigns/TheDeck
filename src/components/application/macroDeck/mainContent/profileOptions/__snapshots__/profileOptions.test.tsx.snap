// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileOptions/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-pVTFL gimsrJ"
        data-testid="profile_options__page-header-wrapper"
      >
        <div
          class="sc-jrQzAO drTyJf"
        >
          <div>
            <input
              class="sc-gsDKAQ sc-crHmcD dEgDrl jpTvIL"
              data-testid="profile_props__profile-name"
              type="text"
              value="Sound Board"
            />
            <select
              class="sc-dkPtRN sc-egiyK iRRDCl cJCtnJ"
              data-testid="profile_props__button-pads"
            >
              <option
                value="6"
              >
                Button Pads: 6
              </option>
              <option
                value="8"
              >
                Button Pads: 8
              </option>
              <option
                value="12"
              >
                Button Pads: 12
              </option>
              <option
                value="15"
              >
                Button Pads: 15
              </option>
              <option
                value="24"
              >
                Button Pads: 24
              </option>
              <option
                value="32"
              >
                Button Pads: 32
              </option>
            </select>
            <button
              class="sc-hKwDye sc-bqiRlB PLLBx gWpitq"
              data-testid="profile_props__submit"
            >
              Submit
            </button>
          </div>
          <div>
            
            <button
              class="sc-hKwDye sc-gKclnd PLLBx iQnRQN"
            >
              New Page
            </button>
            <select
              class="sc-dkPtRN sc-jRQBWg iRRDCl jxmNFY"
              data-testid="page_select__select"
            />
          </div>
        </div>
        <div
          class="sc-kDTinF kazFxB"
        >
          <div
            class="sc-iqseJM jhonZT"
          >
            Settings
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-pVTFL gimsrJ"
      data-testid="profile_options__page-header-wrapper"
    >
      <div
        class="sc-jrQzAO drTyJf"
      >
        <div>
          <input
            class="sc-gsDKAQ sc-crHmcD dEgDrl jpTvIL"
            data-testid="profile_props__profile-name"
            type="text"
            value="Sound Board"
          />
          <select
            class="sc-dkPtRN sc-egiyK iRRDCl cJCtnJ"
            data-testid="profile_props__button-pads"
          >
            <option
              value="6"
            >
              Button Pads: 6
            </option>
            <option
              value="8"
            >
              Button Pads: 8
            </option>
            <option
              value="12"
            >
              Button Pads: 12
            </option>
            <option
              value="15"
            >
              Button Pads: 15
            </option>
            <option
              value="24"
            >
              Button Pads: 24
            </option>
            <option
              value="32"
            >
              Button Pads: 32
            </option>
          </select>
          <button
            class="sc-hKwDye sc-bqiRlB PLLBx gWpitq"
            data-testid="profile_props__submit"
          >
            Submit
          </button>
        </div>
        <div>
          
          <button
            class="sc-hKwDye sc-gKclnd PLLBx iQnRQN"
          >
            New Page
          </button>
          <select
            class="sc-dkPtRN sc-jRQBWg iRRDCl jxmNFY"
            data-testid="page_select__select"
          />
        </div>
      </div>
      <div
        class="sc-kDTinF kazFxB"
      >
        <div
          class="sc-iqseJM jhonZT"
        >
          Settings
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
